arangod/RestHandler/RestReplicationHandler.cpp

```diff
@@ -3061,10 +3061,25 @@ bool RestReplicationHandler::prepareRevisionOperation(
     return false;
   }
 
-  // get resume
-  std::string const& resumeString = _request->value("resume", found);
+  // get resume.
+  // we first try the parameter "resumeHLC" - if set, this will contain a
+  // timestamp-encoded HLC value
+  std::string const& resumeString =
+      _request->value(StaticStrings::RevisionTreeResumeHLC, found);
   if (found) {
-    ctx.resume = RevisionId::fromString(resumeString);
+    // "resumeHLC" is set. use it
+    uint64_t value = HybridLogicalClock::decodeTimeStamp(resumeString);
+    TRI_ASSERT(value != UINT64_MAX);
+    ctx.resume = RevisionId{value};
```

Comment:
Should we add a method `fromHLC` to Revision (similar `fromString`)?

Reply:
Added.